lib/membrane/rtp/packet.ex

```diff
@@ -100,17 +126,39 @@ defmodule Membrane.RTP.Packet do
   def parse(_binary, _parse_payload?), do: {:error, :malformed_packet}
 
   defp parse_header_extension(binary, header_present?)
-  defp parse_header_extension(binary, false), do: {:ok, {nil, binary}}
+  defp parse_header_extension(binary, false), do: {:ok, {[], binary}}
 
   defp parse_header_extension(
-         <<profile_specific::binary-size(2), data_len::16, data::binary-size(data_len)-unit(32),
+         <<_profile_specific::binary-size(2), data_len::16, data::binary-size(data_len)-unit(32),
            rest::binary>>,
          true
        ) do
-    extension = %Header.Extension{profile_specific: profile_specific, data: data}
-
-    {:ok, {extension, rest}}
+    extensions = parse_data_while_possible(data, [])
+    extensions = Enum.reverse(extensions)
+    {:ok, {extensions, rest}}
   end
 
   defp parse_header_extension(_binary, true), do: :error
+
+  defp parse_data_while_possible(<<>>, acc), do: acc
+
+  defp parse_data_while_possible(data, acc) do
+    case parse_one_byte_header(data) do
+      {:ok, {data, rest}} -> parse_data_while_possible(rest, [data | acc])
+      {:ok, rest} -> parse_data_while_possible(rest, acc)
+    end
+  end
+
+  defp parse_one_byte_header(<<extension_header::bits-size(8), extensions::binary>>) do
```

Comment:
An explanation why we don't support two byte header would be appreciated :)

Reply:
According to RFC 8285 section 4.1.2 ", `the one-byte header form is preferred and MUST be supported by all receivers"`. When I was implementing this, I did not need two-byte headers. But I can add it to this PR.